client/src/Console.js

```diff
@@ -0,0 +1,207 @@
+        }
+      })
+      .catch(error => {
+        console.log(error)
+        //this.props.dispatch({ type: 'LOGOUT' });
+      })
+  }
+  toggleRequested(){
+    if(this.props.user_info.requested_to_join_user_ids.length !==
+      this.state.requested_to_join_users.length){
+      this.loadUsers('requested_to_join_users', 
+        this.props.user_info.requested_to_join_user_ids);
+    }
+    
+    this.setState({
+      show_requested_to_join_users: !this.state.show_requested_to_join_users,
+      show_invited_users: false,
+      show_users: false
+    })
+  }
+  toggleInvited(){
+    if(this.props.user_info.invited_user_ids.length !==
+      this.state.invited_users.length){
+      this.loadUsers('invited_users', 
+        this.props.user_info.invited_user_ids);
+    }
+    this.setState({
+      show_requested_to_join_users: false,
+      show_invited_users: !this.state.show_invited_users,
+      show_users: false
+    })
+  }
+  toggleRegular(){
+    if(this.props.user_info.user_ids.length !==
+      this.state.users.length){
+      this.loadUsers('users', this.props.user_info.user_ids);
+    }
+    this.setState({
+      show_requested_to_join_users: false,
+      show_invited_users: false,
+      show_users: !this.state.show_users
+    })
+  }
+  render() {
+    console.log('STATE',this.state)
+    if (this.props.loading){
+      return (<Loading component="Console"/>);
+    }
+    else if (!this.props.logged_in || this.state.redirect_to_root){
+      return (<Redirect to="/" />);
+    }
+    else if (!this.props.user_info.hasOwnProperty('requested_to_join_user_ids')
+      || !this.props.user_info.hasOwnProperty('invited_user_ids')
+      || !this.props.user_info.hasOwnProperty('user_ids')){
+      return (<Loading component={'Users'}/>);
+    }
+    else if(!this.props.user_info.is_supervisor || 
+      this.props.user_info.managed_groups_ids.indexOf(
+        this.props.match.params.group_id) === -1){
+      return (<Redirect to="/" />);
+    }
+    else {
+      console.log('info',this.props.user_info)
+      return (
+      <div style={{padding:'2em'}}>
+      <Jumbotron style={{minHeight:'27em', paddingTop:'0', 
+        textAlign:'center'}}>
+        <div style={{paddingBottom:'1em'}}>
+          <h4>Administrator Console {' '}</h4>
+          <h5><Badge pill color="primary">
+            {this.props.user_info.group_name}</Badge></h5>
+        </div>
+
+        <ButtonGroup>
+          <Button color="primary" onClick={this.toggleRequested}
+            active={this.state.show_requested_to_join_users}>
+            Show Requested Users</Button>
+          <Button color="primary" onClick={this.toggleInvited}
+            active={this.state.show_invited_users}>
+            Show Invited Users</Button>
+          <Button color="primary" onClick={this.toggleRegular}
+            active={this.state.show_users}>
+            Show Joined Users</Button>
+        </ButtonGroup>
+
+        <ConsoleUsersListGroup

```

Comment:
can potentially use the spread operator here so we don't have to pass all these props and state vars in.
example: https://www.techiediaries.com/react-spread-operator-props-setstate/

Reply:
Thing is that would pass in some extra unnecessary props and the way it is right now allows easier notation in ConsoleUsersListGroup because user_info.user_ids becomes just user_ids, user_info.invited_user_ids becomes just invited_user_ids, etc.

Reply:
makes sense, just want to try to find a way to reduce the number of props we are passing in if possible. could we only pass in `invited_users` instead of both `invited_users` and `invited_users_ids` since we should easily be able to find the id from the user? we could do this for some of the other props too